sdk/core/azure-core/src/main/java/com/azure/core/http/rest/Page.java

```diff
@@ -3,26 +3,28 @@
 
 package com.azure.core.http.rest;
 
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.paging.ContinuablePage;
+
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Represents a paginated REST response from the service.
  *
- * @param <T> Type of the listed objects in that response.
+ * @param <T> Type of items in the page response.
  */
-public interface Page<T> {
-
-    /**
-     * Gets a list of items returned from the service.
-     *
-     * @return A list of items from the service.
-     */
-    List<T> getItems();
-
+public interface Page<T> extends ContinuablePage<String, T> {
     /**
-     * Gets a link to the next page, or {@code null} if there are no more results.
+     * @return list of elements in the page.
      *
-     * @return A link to the next page, or {@code null} if there are no more results.
+     * @deprecated use {@link this#getElements()}.
      */
-    String getContinuationToken();
+    @Deprecated
+    default List<T> getItems() {
+        IterableStream<T> iterableStream = this.getElements();
+        return iterableStream == null
+            ? null
```

Comment:
@JonathanGiles Do we have a guideline about when to return an empty collection and when to return null? I noticed this also in: https://github.com/Azure/azure-sdk-for-java/pull/6465/files#diff-f9bbb7ca67fb43df109d0ca82503fb7fR29 and a few other places.

I personally prefer returning an empty collection, because you'd know that there are no items, whereas, you'd have to do a null check, then check for empty if we return `null`.

Reply:
I don't think I have anything in the guidelines, but I strongly prefer to always return an empty collection rather than null.

Reply:
agreed, it is good practice that our APIs exposing collection type to return EMPTY collection instead of NULL. The SDK author implementing `Page` or `ContinuablePage` MUST return an EMPTY IterableStream from `getElements()`, not NULL. The API we're discussing here is the default impl of Deprecated `getItems()` which uses `getElements()` underneath, so as long as SDK author implementing `getElements()` returns EMPTY stream, the `getItems()` will also return EMPTY list.

Reply:
updated to return empty list instead of null